src/components/sidebar/LineItemSubMenu.tsx

```diff
@@ -49,31 +59,36 @@ class LineItemSubMenu extends Component<LineItemSubMenuProps, LineItemSubMenuSta
             });
     }
 
-    private select(index: number) {
+    private select(routePathId: string) {
+        const newSelectedIds = this.state.selectedIds;
+        newSelectedIds.push(routePathId);
         this.setState({
-            checkedRoutePaths: this.state.checkedRoutePaths.concat(index),
+            selectedIds: newSelectedIds,
         });
+        this.props.searchStore!.addSubLineItem(this.props.routeId, routePathId);
```

Comment:
We should probably avoid using both store and local state, when the data we handle basically is the same. What if the data only would be in the store, and a routePath could check directly from there if it is checked or not?